src/tqec/sketchup/zx_graph.py

```diff
@@ -0,0 +1,334 @@
+"""ZX graph representation of a 3D spacetime defect diagram."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
```

Comment:
Needed for typing errors below
```suggestion
from typing import TYPE_CHECKING, cast
```

Reply:
Disregard that, I will do a proper review when you think the PR is ready :)